pylint/checkers/classes/class_checker.py

```diff
@@ -1999,11 +1999,12 @@ def _check_init(self, node: nodes.FunctionDef, klass_node: nodes.ClassDef) -> No
 
             # Return if any of the klass' first-order bases is protocol
             for base in klass.bases:
-                # We don't need to catch InferenceError here as _ancestors_to_call
-                # already does this for us.
-                for inf_base in base.infer():
```

Comment:
Should we use ``safe_infer`` and a ``is not None`` check instead ? if

Reply:
The problem of using `safe_infer` is that it only returns one node. Sometimes I don't care if there is ambiguity about the nodes, if I can recognise one of them as being a typing protocol (as I do in this case) I want to exit. `safe_infer` doesn't really allow that.

Reply:
Ho I thought it was equivalent but with all exceptions caught, thanks..